openmc/openmoc_compatible.py

```diff
@@ -0,0 +1,825 @@
+
+    # Create an OpenMC Material to represent this OpenMOC Material
+    name = openmoc_material.getName()
+    openmc_material = openmc.Material(material_id=material_id, name=name)
+
+    # Add the OpenMOC Material to the global collection of all OpenMOC Materials
+    OPENMOC_MATERIALS[material_id] = openmoc_material
+
+    # Add the OpenMC Material to the global collection of all OpenMC Materials
+    OPENMC_MATERIALS[material_id] = openmc_material
+
+    return openmc_material
+
+
+def get_openmoc_surface(openmc_surface):
+    """Return an OpenMOC surface corresponding to an OpenMC surface.
+
+    Parameters
+    ----------
+    openmc_surface : openmc.Surface
+        OpenMC surface
+
+    Returns
+    -------
+    openmoc_surface : openmoc.Surface
+        Equivalent OpenMOC surface
+
+    """
+
+    cv.check_type('openmc_surface', openmc_surface, openmc.Surface)
+
+    surface_id = openmc_surface.id
+
+    # If this Material was already created, use it
+    if surface_id in OPENMOC_SURFACES:
+        return OPENMOC_SURFACES[surface_id]
+
+    # Create an OpenMOC Surface to represent this OpenMC Surface
+    name = openmc_surface.name
+
+    # Determine the type of boundary conditions applied to the Surface
+    if openmc_surface.boundary_type == 'vacuum':
+        boundary = openmoc.VACUUM
+    elif openmc_surface.boundary_type == 'reflective':
+        boundary = openmoc.REFLECTIVE
+    elif openmc_surface.boundary_type == 'periodic':
+        boundary = openmoc.PERIODIC
+    else:
+        boundary = openmoc.BOUNDARY_NONE
+
+    if openmc_surface.type == 'plane':
+        A = openmc_surface.a
+        B = openmc_surface.b
+        C = openmc_surface.c
+        D = openmc_surface.d
+
+        # OpenMOC uses the opposite sign on D
+        openmoc_surface = openmoc.Plane(A, B, C, -D, surface_id, name)
+
+    elif openmc_surface.type == 'x-plane':
+        x0 = openmc_surface.x0
+        openmoc_surface = openmoc.XPlane(x0, surface_id, name)
+
+    elif openmc_surface.type == 'y-plane':
+        y0 = openmc_surface.y0
+        openmoc_surface = openmoc.YPlane(y0, surface_id, name)
+
+    elif openmc_surface.type == 'z-plane':
+        z0 = openmc_surface.z0
+        openmoc_surface = openmoc.ZPlane(z0, surface_id, name)
+
+    elif openmc_surface.type == 'z-cylinder':
+        x0 = openmc_surface.x0
+        y0 = openmc_surface.y0
+        R = openmc_surface.r
+        openmoc_surface = openmoc.ZCylinder(x0, y0, R, surface_id, name)
+
+    else:
+        msg = 'Unable to create an OpenMOC Surface from an OpenMC ' \
+              'Surface of type "{}" since it is not a compatible ' \
+              'Surface type in OpenMOC'.format(type(openmc_surface))
+        raise ValueError(msg)
+
+    # Set the boundary condition for this Surface
+    openmoc_surface.setBoundaryType(boundary)
+
+    # Add the OpenMC Surface to the global collection of all OpenMC Surfaces
+    OPENMC_SURFACES[surface_id] = openmc_surface
```

Comment:
Maybe a good idea to clarify the naming conventions between `OPENMC_SURFACES` and `OPENMOC_SURFACES`. At first glance they are extremely identical and might cause future issues if someone is new to the code base.